bolt/lib/Core/BinaryFunction.cpp

```diff
@@ -2462,6 +2462,13 @@ struct CFISnapshot {
     case MCCFIInstruction::OpDefCfaRegister:
       CFAReg = Instr.getRegister();
       CFARule = UNKNOWN;
+
+      // This shouldn't happen according to the spec but GNU binutils on RISC-V
+      // emits a DW_CFA_def_cfa_register in CIE's which leaves the offset
+      // unspecified. Both readelf and llvm-dwarfdump interpret the offset as 0
+      // in this case so let's do the same.
```

Comment:
I did a quick check and this indeed seems to be the case:
- libunwind initializes all CFI columns to 0 [here](https://github.com/libunwind/libunwind/blob/24dc3b0880695137a48c2ad0244900f9bb8416eb/src/dwarf/Gparser.c#L521-L523) (CFA offset is stored in a special column).
- libgcc sets most of the state to 0 in a confusing `memset` [here](https://gcc.gnu.org/git/?p=gcc.git;a=blob;f=libgcc/unwind-dw2.c;h=d0afce7a9ea9f5b12a5a01ef1e940e1452b48cab;hb=HEAD#l997). Since `regs.cfa_offset` comes after `regs.how`, it is set to 0 there.

So both runtimes default CFA offsets to zero. Although it seems to be more of a coincident than a deliberate choice in both cases.